src/systemctl/systemctl.c

```diff
@@ -5272,6 +5272,17 @@ static int cat(int argc, char *argv[], void *userdata) {
                 else
                         puts("");
 
+                if (need_daemon_reload(bus, *name))
+                        printf("%s# %sWarning:%s %s changed on disk, showing current disk contents.\n"
```

Comment:
> Ha, I thought that this would be a more complicated patch. It's nice when existing functionality gets reused like that.

Yes, I’m very glad I didn’t have to add a new dbus method to return unit files + time loaded or something like that :)

> Why make the message blue? Shouldn't it be red as other warnings?

The word “Warning” is still red, just as in `systemctl status`. I only changed the color of the rest of the text from default to blue, to match the file name comments.

> The first line basically says that you're showing the latest version ;)

I’m not sure what you mean… `systemctl cat` really is `cat`, it boils down to `open(filename)` and `copy_bytes(fd, STDOUT_FILENO)` (see function `cat_file`), showing whatever is currently on disk, not what the daemon loaded. Are you saying the message sounds like it’s showing what the daemon loaded? Then I should clarify it.

There’s another condition to consider, which exhibits the same behavior as the second case.
- a file was added that replaces the main unit file (`/etc` vs. `/usr/lib`, `~/.config` vs. `/etc`, etc.)

This case is also not detected by `need_daemon_reload`.

I’m not sure if the message should reflect both cases – unless I’m missing something, we won’t even print the message in the second case, since we don’t know it’s happening.